app/controllers/api/v1/coupons_controller.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class CouponsController < Api::BaseController
+      def create
+        service = Coupons::CreateService.new
+        result = service.create(
+          **input_params
+            .merge(organization_id: current_organization.id)
+            .to_h
+            .symbolize_keys
+        )
+
+        if result.success?
+          render(
```

Comment:
Since this exact same code is used in `show`, `create`, `update` and `destroy`, I'm wondering if we could add a private shared `render_coupon` method. What do you think?

Reply:
Sounds great to me! Refactored